src/addon/addon_manager.cpp

```diff
@@ -645,7 +675,12 @@ AddonManager::add_installed_archive(const std::string& archive, const std::strin
   {
     std::string os_path = FileSystem::join(realdir, archive);
 
-    PHYSFS_mount(os_path.c_str(), nullptr, 1);
+ if (g_config->allow_local_files) {
+    PHYSFS_mount(os_path.c_str(), nullptr, 0);
+    } else {
+ PHYSFS_mount(os_path.c_str(), nullptr, 1);
+    }
+
```

Comment:
This can be simplified to `PHYSFS_mount(os_path.c_str(), nullptr, !(g_config->allow_local_files));`